energy/energy_generation_consumption.py

```diff
@@ -4,22 +4,50 @@
 
 # COMMAND ----------
 
-# Bank firewall prenvents programmatic data fetch: https://ember-climate.org/data/api/
-# Curretnly requesting ITS to whitelist this request
+import requests
 
-file_path = 'https://raw.githubusercontent.com/weilu/mega-indicators/main/energy/2023-Country-Transition-Tracker-Consolidated-Format.csv'
-raw_df = pd.read_csv(file_path)
+my_api_key = dbutils.secrets.get(scope="DIMEBOOSTKEYVAULT", key="ember_energy_key")
+base_url = "https://api.ember-climate.org"
+query_url = (
+    f"{base_url}/v1/electricity-generation/yearly"
+    + f"?is_aggregate_series=false&api_key={my_api_key}"
+)
+
+response = requests.get(query_url)
+
+if response.status_code == 200:
+    data = response.json()
+
+# The renaming is necessary to preserve the consistency with the published energy data https://ember-climate.org/data-catalogue/yearly-electricity-data/
+raw_df = pd.DataFrame(data["data"]).rename(
+    columns={
+        "date": "Year",
+        "entity_code": "Country code",
+        "series": "Variable",
+        "entity": "Area",
+        "share_of_generation_pct": "Value"
+    },
+)
```

Comment:
I actually think we should consistently use snake_case for column names as our other indicator datasets are.

Reply:
fixed!